### ISSUE TITLE
PyPy-specific: `ctypes.Structure` fields not ignored; `no-member` false positive

### ISSUE THREAD

**Issue by emcd** [PyCQA/pylint#7527]
### Bug description

Note: This bug is specific to running on Pylint on PyPy; it is not reproducible on CPython.

Pylint's handling of `ctypes.Structure` is not implementation-agnostic. It apparently references the `ctypes` implementation of the platform on which it is running when analyzing `ctypes`-using code that may not even be intended for that platform.

For the following reproducer, the Pylint scan passes cleanly, when running on CPython, but fails with a `no-member` error, when running on PyPy:
```python
''' Reproducer for false positive on PyPy-based Pylint. '''

import sys
from ctypes import Structure, c_ssize_t, c_void_p
from sys import implementation

class Class( type ): ''' Trivial metaclass. '''
class Object: ''' Trivial class. '''

if 'cpython' == implementation.name and not hasattr( sys, 'getobjects' ):
    class PyObject( Structure ):
        ''' Structural representation of normal :c:struct:`PyObject`. '''
        _fields_ = (
            ( 'ob_refcnt', c_ssize_t ),
            ( 'ob_type', c_void_p )
        )
    new_metaclass_pointer = id( Class )
    class_struct = PyObject.from_address( id( Object ) )
    class_struct.ob_type = c_void_p( new_metaclass_pointer )
    new_metaclass_struct = PyObject.from_address( new_metaclass_pointer )
    new_metaclass_struct.ob_refcnt += 1  # On PyPy, complains about 'no-member'

    print( type( Object ) )
```

### Command used

```shell
pylint --reports=no --score=no pylint_ctypes_bug.py
```

### Pylint output

```shell
************* Module pylint_ctypes_bug
pylint_ctypes_bug.py  21, 4 [no-member] Instance of 'StructOrUnion' has no 'ob_refcnt' member
```

### Expected behavior

No output, just like when it runs on CPython.

### Pylint version

```shell
CPython:

pylint 2.15.3
astroid 2.12.10
Python 3.7.14 (default, Sep 12 2022, 16:42:27)
[GCC 9.4.0]
```
PyPy:
```
pylint 2.15.3
astroid 2.12.10
Python 3.8.13 (4b1398fe9d76ad762155d03684c2a153d230b2ef, Mar 29 2022, 07:08:24)
[PyPy 7.3.9 with GCC 10.2.1 20210130 (Red Hat 10.2.1-11)]
```
```


### OS / Environment

_No response_

### Additional dependencies

_No response_
